src/platform/plugins/shared/workflows_execution_engine/server/step/on_failure/README.md

```diff
@@ -0,0 +1,113 @@
+# On-Failure Configuration
+
+The `on-failure` configuration provides error handling for workflow steps. It can be configured at the step level or workflow level (in `settings`).
+
+
+## Configuration Levels
+
+### Step Level
+
+```yaml
+steps:
+  - name: api-call
+    type: http
+    on-failure:
+      retry:
+        max-attempts: 3
+        delay: "5s"
+```
+
+### Workflow Level
+
+```yaml
+settings:
+  on-failure:
+    retry:
+      max-attempts: 2
+      delay: "1s"
+steps:
+  - name: api-call
+    type: http
+```
+
+**Note:** Step-level `on-failure` takes precedence over workflow-level configuration.
+
+## Options
+
+### Retry
+
+Retries the step on failure with configurable attempts and delay.
+
+```yaml
+on-failure:
+  retry:
+    max-attempts: 3  # Required: minimum 1
+    delay: "5s"      # Optional: duration format (e.g., "5s", "1m", "2h")
+```
+
+**Behavior:**
+- Retries up to `max-attempts` times
+- Waits `delay` between retries (if specified)
+- Workflow enters `WAITING` state if delay exceeds task manager threshold
+- Workflow fails if all retries are exhausted
+
+### Fallback
+
+Executes alternative steps when the primary step fails.
+
+```yaml
+on-failure:
+  fallback:
+    - name: fallback-step
+      type: http
+      with:
+        url: "https://backup-api.example.com"
+```
+
+**Behavior:**
+- Executes fallback steps only after retries are exhausted (if retry is configured)
+- Original error is preserved in workflow execution
+- Workflow continues after fallback execution
```

Comment:
No, in the current implementation it will continue only if the user specifies "continue: true".
Without "continue: true" it the execution still fails even though fallback is specified.

The pseudocode of fallback behavior:
```ts
try {
   primaryPath();
} catch {
   fallbackPath();
   throw;
}
```